### PR TITLE
bugfix - bakers table with empty cycle and period[backport]

### PR DESCRIPTION
**Pull request by piotrkosecki**
fixed the issue where in bakers table could contain empty cycle/period for non genesis block

### CODE REVIEW

File: src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala [hunk 1]
```diff
@@ -704,9 +704,13 @@ object DatabaseConversions extends LazyLogging {
...
   implicit val blockAccountsAssociationToCheckpointRow =
-    new Conversion[List, (BlockHash, Int, Option[Instant], Option[Int], List[AccountId]), Tables.AccountsCheckpointRow] {
-      override def convert(from: (BlockHash, Int, Option[Instant], Option[Int], List[AccountId])) = {
-        val (blockHash, blockLevel, timestamp, cycle, ids) = from
+    new Conversion[
+      List,
+      (BlockHash, Int, Option[Instant], Option[Int], Option[Int], List[AccountId]),
+      Tables.AccountsCheckpointRow
+    ] {
+      override def convert(from: (BlockHash, Int, Option[Instant], Option[Int], Option[Int], List[AccountId])) = {
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala:712]: I won't block the PR for this, but I honestly start to find it hard to make sense of so many opaque parameters, with no structure and totally common types.
I personally feel it gets hard to follow and prone to easily make mistake.

I leave it to you if there's anything to improve or not.

File: src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala [hunk 2]
```diff
@@ -350,7 +350,7 @@ object TezosDatabaseOperations extends LazyLogging {
         ids =>
           writeAccountsCheckpoint(
             List(
-              (hash, level, Some(timestamp), cycle, ids.map(AccountId(_)).toList)
+              (hash, level, Some(timestamp), cycle, None, ids.map(AccountId(_)).toList)
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala:353]: aren't we losing info here?

File: src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala (hunk 1 above)
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala:712]: I think next time when I'll be out of issues I'll improving this, I don't think it's necessary for the bugfix PR

File: src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala (hunk 2 above)
Comment (ivanopagano) [reply src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala:353]: ok,but I can't clearly make out, why we need to pass None if that parameter is not needed anyway? Can't it be removed completely from the expected values at this point in the code?

Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala:353]: but in one place we use `.asTuple` to extract this data for passing to the `writeAccountsCheckpoint`, and here we're not.

Comment (ivanopagano) [reply src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala:353]: ok, now my eyes are opened... I can see clearly :eye:
